ca_gazebo/include/ca_gazebo/color_sensor_plugin.h

```diff
@@ -0,0 +1,48 @@
+    #ifndef GAZEBO_ROS_COLOR_SENSOR_HH
+    #define GAZEBO_ROS_COLOR_SENSOR_HH
+    
+    #include <string>
+    
+    // library for processing camera data for gazebo / ros conversions
+    #include <gazebo/plugins/CameraPlugin.hh>
+
+
+    #include <gazebo_plugins/gazebo_ros_camera_utils.h>
+    
+
+    namespace gazebo
+    {
+      class GazeboRosColor : public CameraPlugin, GazeboRosCameraUtils
+      {
+        /// \brief Constructor
+        /// \param parent The parent entity, must be a Model or a Sensor
+        public: GazeboRosColor();
+    
+        /// \brief Destructor
+        public: ~GazeboRosColor();
+    
+        /// \brief Load the plugin
+        /// \param take in SDF root element
+        public: void Load(sensors::SensorPtr _parent, sdf::ElementPtr _sdf);
+    
+        /// \brief Update the controller
+        protected: virtual void OnNewFrame(const unsigned char *_image,
+        unsigned int _width, unsigned int _height,
+        unsigned int _depth, const std::string &_format);
+        unsigned int _rvalue_max;
+        unsigned int _rvalue_min;
+        unsigned int _gvalue_max;
+        unsigned int _gvalue_min;
+        unsigned int _bvalue_max;
+        unsigned int _bvalue_min; 
+        std::string _publisher_name;
+        unsigned int _factor_color_per_image;
+
+        ros::NodeHandle _nh;
+        ros::Publisher _sensorPublisher;
```

Comment:
Are you going to use camelCase or snake_case?

Reply:
Class data members in C++ are named differently than in Python. The underscore goes in the end. See here:
https://google.github.io/styleguide/cppguide.html#Variable_Names